sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -74,12 +74,23 @@ class QueryExecution(
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
 
+  lazy val commandCollected: LogicalPlan = analyzed mapChildren { child =>
+    child transform {
+      // SPARK-35378: Eagerly execute Command so that query command with CTE
+      case c: Command =>
+        val subQueryExecution = sparkSession.sessionState.executePlan(c)
+        LocalRelation(c.output,
+          subQueryExecution.executedPlan.executeCollect(), false, Some(subQueryExecution.id))
```

Comment:
How about we create new query plan nodes: `CommandResult` and `CommandResultExec`
```
case class CommandResult(qe: QueryExecution) extends LeadNode {
  def innerChildren = Seq(qe.analyzedPlan)
}
case class CommandResultExec(qe: QueryExecution) extends LeafNodeExec {
  def innerChildren = Seq(qe.executedPlan)
}
```

then both UI and EXPLAIN can have pretty output.